streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -505,27 +506,27 @@ public Subscription subscription(Set<String> topics) {
             final Set<String> consumers = entry.getValue().consumers;
             final ClientState state = entry.getValue().state;
 
-            final ArrayList<TaskId> taskIds = new ArrayList<>(state.assignedTaskCount());
-            final int numActiveTasks = state.activeTaskCount();
+            final List<List<TaskId>> interleavedActive = interleaveTasksByGroupId(state.activeTasks(), consumers.size());
+            final List<List<TaskId>> interleavedStandby = interleaveTasksByGroupId(state.standbyTasks(), consumers.size());
```

Comment:
If we call this twice and build two list, we cannot guarantee load balanced assignment across threads. Assume you have two thread and 3 active and 3 standby tasks. We would end up with:
```
t1: a0, a2, s0, s2
t2: a1, s1
```
However, after assigning all active tasks, we should not start to assign standby tasks on the first thread but on the "next" thread:
```
t1: a0, a2, s1
t2: a1, s0, s2
```
We can achieve this, by building a single list of active and standby tasks IMHO.

Or do you think that this kind of load balancing is secondary? \cc @guozhangwang @dguy

Reply:
I think this could work taking a look